DeusEx/Classes/DXRSkills.uc

```diff
@@ -33,13 +33,21 @@ function RandoSkills()
 {
     local Skill aSkill;
     local int i, m;
-    local int percent;
+    local int percent, mission_group;
     local float f;
     local SkillCostMultiplier scm;
     local class<Skill> c;
 
-    l("randomizing skills with seed " $ dxr.seed $ ", min: "$dxr.flags.minskill$", max: "$dxr.flags.maxskill);
-    dxr.SetSeed(dxr.seed);
+    l("randomizing skills with seed " $ dxr.seed $ ", min: "$dxr.flags.minskill$", max: "$dxr.flags.maxskill $", reroll_missions: "$ dxr.flags.skills_reroll_missions $", independent_levels: "$ dxr.flags.skills_independent_levels );
+    if( dxr.flags.skills_reroll_missions == 0 )
+        dxr.SetSeed(dxr.seed);
+    else {
+        if( dxr.dxInfo != None )
+            mission_group = dxr.dxInfo.missionNumber;
+        mission_group = Clamp(mission_group, 1, 1000) / dxr.flags.skills_reroll_missions;
```

Comment:
mission 0 or none becomes 1, that way you won't look through skills before starting the game, and then immediately look at the skills again once you get into the game